chore: added new achitecture page with diagram (#333)
LedgerHQ/developer-portal: pages/docs/ledger-live/exchange/sell/architecture.mdx

## Workflow Diagram

```mermaid
sequenceDiagram
    participant User
    participant LedgerLive as Ledger Live
    participant LedgerBackend as Ledger Backend
    participant ProviderBackend as Provider Backend
    participant ProviderWidget as Provider Widget App
    participant ExchangeSDK as Exchange SDK

    %% Step 1: Open Sell App
    User->>LedgerLive: Open Sell App
    LedgerLive->>LedgerBackend: [GET] /providers
    LedgerBackend->>ProviderBackend: [GET] /capabilities (Cached)
    ProviderBackend-->>LedgerBackend: Return capabilities data
    LedgerBackend->>ProviderBackend: [GET] /crypto-currencies (Cached)
    ProviderBackend-->>LedgerBackend: Return supported crypto assets
    LedgerBackend-->>LedgerLive: Return providers list and capabilities

    %% Step 2: User Input and Validation
    User->>LedgerLive: Select crypto, fiat, amount
    LedgerLive-->>User: Validate input using cached /capabilities
    User->>LedgerLive: Proceed with valid input

>     LedgerLive-->>User: Display countries and payment methods based on /capabilities
> 
>     %% Step 3: Select Provider and Get Quotes
>     User->>LedgerLive: Select country and payment method

Reviewer: ```suggestion
    LedgerLive-->>User: Display countries and payment methods based on /capabilities
    User->>LedgerLive: Select country and payment method
    
    %% Step 3: Select Provider and Get Quotes
```

    LedgerLive->>LedgerBackend: [GET] /offRamp Find eligible providers
    LedgerBackend->>ProviderBackend: [GET] /quotes
    ProviderBackend-->>LedgerBackend: Return quotes
    LedgerBackend-->>LedgerLive: Return providers and quotes
    User->>LedgerLive: Choose provider

    %% Step 4: Redirect to Provider Widget
    LedgerLive->>ProviderWidget: Redirect with parameters (LiveApp URL)

>     User->>ProviderWidget: Landing in step transaction in provider widget

Reviewer: Idem?
```suggestion
    User->>ProviderWidget: KYC process, if required, and landing in step transaction in provider widget.
```

## Workflow Diagram

<Zoom>
    <img
      alt="Flow Diagram"
      src="/exchange/sell-flow-diagram.png"
    />
</Zoom>

### Step 1: Open Sell App

<b>Action:</b> The user opens the Sell App in Ledger Live.

<b>Ledger Live:</b>

- Fetches capabilities data from the Ledger Backend, which caches responses from the provider's [/capabilities](https://exchange-integration-sell.redoc.ly/#operation/getCapabilities) endpoint.

- Fetches supported cryptocurrencies from the Ledger Backend, which caches responses from the provider's [/crypto-currencies](https://exchange-integration-sell.redoc.ly/#operation/getCryptoCurrencies) endpoint.

### Step 2: User Input and Validation

<b>User Interaction:</b> The user selects the cryptocurrency, fiat currency, and the amount they wish to sell.

<b>Validation Process:</b> Ledger Live validates user input.
Validation is performed using cached data from [/capabilities](https://exchange-integration-sell.redoc.ly/#operation/getCapabilities), ensuring the user's input aligns with provider-supported limitations (e.g., minimum/maximum amounts, supported countries).

If the input is invalid, Ledger Live displays an error message to the user.

> ### Step 3: Select Provider and Get Quotes

Reviewer: ```suggestion
### Step 3: Get Quotes and Select Provider 
```

> ### Step 2: User Input and Validation
> 
> <b>User Interaction:</b> The user selects the cryptocurrency, fiat currency, and the amount they wish to sell.
> 
> <b>Validation Process:</b> Ledger Live validates user input.
> Validation is performed using cached data from [/capabilities](https://exchange-integration-sell.redoc.ly/#operation/getCapabilities), ensuring the user's input aligns with provider-supported limitations (e.g., minimum/maximum amounts, supported countries).
> 
> If the input is invalid, Ledger Live displays an error message to the user.
> 
> ### Step 3: Select Provider and Get Quotes
> 
> <b>User Interaction:</b> The user selects their country and preferred payment method (e.g., PayPal, bank transfer).

Reviewer: ```suggestion
### Step 2: User Input and Validation

<b>User Interaction:</b> The user selects the cryptocurrency, fiat currency, and the amount they wish to sell.

<b>Validation Process:</b> Ledger Live validates user input.
Validation is performed using cached data from [/capabilities](https://exchange-integration-sell.redoc.ly/#operation/getCapabilities), ensuring the user's input aligns with provider-supported limitations (e.g., minimum/maximum amounts, supported countries).

If the input is invalid, Ledger Live displays an error message to the user.

### Step 3: Get Quotes and Select Provider

<b>User Interaction:</b> The user request quotes.
```

The user lands on the provider widget to complete the transaction details.

### Step 5: Initiate Transaction

<b>Provider Widget Interaction:</b> The provider widget initializes the transaction by calling the `exchangeSdk.sell` method.

This triggers the Ledger Backend to initiate the transaction.

<b>Backend Process:</b>

Ledger Backend sends a [POST] [/sell](https://exchange-integration-sell.redoc.ly/#operation/postSell) request to the provider backend, requesting a signed payload for the transaction.

The provider backend processes the request and returns the signed payload.

Ledger Backend passes the signed payload to the Exchange SDK.

<b>Device Approval:</b>

The Exchange SDK prompts the Ledger device to approve or reject the transaction.

The user approves or rejects the transaction on their Ledger device.

### Step 6: Finalizing the Transaction

<b>Post-Approval:</b> Once the transaction is approved, the Ledger Backend informs the provider backend of the transaction's approval, providing a transactionId if necessary.

> If the transaction is rejected, the provider is also informed.

Reviewer: Is this true? I do not see this line in the diagram.
In the diagram exchangeSDK promise is responding with this info to provider widget. And provider widget must inform provider backend.

Author: I meant by this that exchangeSdk responds with error or success(transactionId)
<img width="1244" alt="image" src="https://github.com/user-attachments/assets/077e55f5-efa3-42bd-8cea-a5f3bf6c2cc0" />

Reviewer: Then Ledger Backend do NOT informs the provider backend of the transaction's approval.

```suggestion
<b>Post-Approval:</b> After the transaction is approved, the exchangeSDK.sell promise notifies the provider widget about the transaction's approval, including a transactionId if approved. It is the responsibility of the provider widget to inform the provider's backend of the final transaction status to initiate the webhook if the user has approved the transaction.

In the event the transaction is rejected, the provider is similarly notified.
```

<b>Redirection Details:</b> The redirection includes required parameters as specified in the [LiveApp URL documentation](../sell/providers-liveapp), such as the selected cryptocurrency, fiat currency, amount, and user-specific details.

The user lands on the provider widget to complete the transaction details.

### Step 5: Initiate Transaction

<b>Provider Widget Interaction:</b> The provider widget initializes the transaction by calling the `exchangeSdk.sell` method.

This triggers the Ledger Backend to initiate the transaction.

<b>Backend Process:</b>

Ledger Backend sends a [POST] [/sell](https://exchange-integration-sell.redoc.ly/#operation/postSell) request to the provider backend, requesting a signed payload for the transaction.

The provider backend processes the request and returns the signed payload.

Ledger Backend passes the signed payload to the Exchange SDK.

<b>Device Approval:</b>

The Exchange SDK prompts the Ledger device to approve or reject the transaction.

The user approves or rejects the transaction on their Ledger device.

### Step 6: Finalizing the Transaction

> <b>Post-Approval:</b> After the transaction is approved, the exchangeSDK.sell promise notifies the provider widget about the transaction's approval, including a transactionId if approved. It is the responsibility of the provider widget to inform the provider's backend of the final transaction status to initiate the webhook if the user has approved the transaction.

Reviewer 2: Supported

> In the event the transaction is rejected, the provider is similarly notified.

Reviewer: ```suggestion
If the event the transaction is rejected, the provider is similarly notified.
```

Reviewer 2: Supported
